homeassistant/components/daikin/sensor.py

```diff
@@ -74,3 +106,55 @@ async def async_update(self):
     def device_info(self):
         """Return a device description for device registry."""
         return self._api.device_info
+
+
+class DaikinClimateSensor(DaikinSensor):
+    """Representation of a Climate Sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_TEMPERATURE
+
+    @property
+    def state(self):
+        """Return the internal state of the sensor."""
+        if self._device_attribute == ATTR_INSIDE_TEMPERATURE:
+            return self._api.device.inside_temperature
+        if self._device_attribute == ATTR_OUTSIDE_TEMPERATURE:
+            return self._api.device.outside_temperature
+        return None
+
+
+class DaikinPowerSensor(DaikinSensor):
+    """Representation of a power/energy consumption sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self._device_attribute == ATTR_TOTAL_POWER:
+            return POWER_KILO_WATT
+        if self._device_attribute in (ATTR_COOL_ENERGY, ATTR_HEAT_ENERGY):
+            return ENERGY_KILO_WATT_HOUR
```

Comment:
The power sensor entity only supports W or kW as units.

https://developers.home-assistant.io/docs/entity_sensor#available-device-classes

Reply:
I've replaced by kW for the 3 sensors even though kWh would have been more suitable

Reply:
The `total_power` sensor measures the total **instant** power consumption (kW is the right unit) but the `cool_energy` and `heat_energy` can only measure the power consumption **over the last hour** (kWh would have been better)

Reply:
We could make a third sensor class for the energy consumption without device class and set the correct unit.

Reply:
Using kW is not wrong either as 1 kWh = 1 kW over 1 hour, it just looks like it is an instant power measure even though it is not. But I think I prefer to fit the generic home assistant' sensor classes to reduce the PR complexity

Reply:
1 kWh = 3,6×10^+6 Joules, and that is energy.

1 kW = power (actually defined as: "the SI unit of power, equivalent to one joule per second, corresponding to the rate of consumption of energy in an electric circuit where the potential difference is one volt and the current one ampere.")

So energy != power. 

I think they should be different.

Reply:
Ok I've removed the power device class from the cool/heat energy sensors